src/Draggable/Sensors/DragSensor/tests/DragSensor.test.js

```diff
@@ -0,0 +1,514 @@
+      document.elementFromPoint = () => draggable;
+
+      // Fire dragstart event to initialize dragging state
+      triggerEvent(draggable, 'dragstart', {
+        dataTransfer: getDataTransferStub(),
+      });
+
+      // Fire primary dragend event
+      triggerEvent(draggable, 'dragend');
+
+      expect(getSensorEventsByType('drag:stop').length)
+        .toBe(1);
+    });
+
+    test('should configure DragStopSensorEvent correctly', () => {
+      const draggable = sandbox.querySelector('li');
+      const expectedClientX = 15;
+      const expectedClientY = 984;
+      document.elementFromPoint = () => draggable;
+
+      // Fire dragstart to initialize dragging state
+      triggerEvent(draggable, 'dragstart', {
+        dataTransfer: getDataTransferStub(),
+      });
+
+      // Fire primary dragend event
+      const dispatchedBrowserEvent = triggerEvent(draggable, 'dragend', {
+        clientX: expectedClientX,
+        clientY: expectedClientY,
+      });
+
+      const expectedContainer = document.querySelectorAll('ul')[0];
+      const actualDragStopSensorEvent = getSensorEventsByType('drag:stop')[0];
+
+      expect(actualDragStopSensorEvent.clientX)
+        .toBe(expectedClientX);
+
+      expect(actualDragStopSensorEvent.clientY)
+        .toBe(expectedClientY);
+
+      expect(actualDragStopSensorEvent.container)
+        .toBe(expectedContainer);
+
+      expect(actualDragStopSensorEvent.originalEvent)
+        .toBe(dispatchedBrowserEvent);
+    });
+
+    test('should prevent default on `DragEnd` event', (done) => {
+      const draggable = sandbox.querySelector('li');
+      document.elementFromPoint = () => draggable;
+
+      // Fire dragstart event to initialize dragging state
+      triggerEvent(draggable, 'dragstart', {
+        dataTransfer: getDataTransferStub(),
+      });
+
+      // Fire primary dragend event
+      const dispatchedBrowserEvent = triggerEvent(draggable, 'dragend');
+
+      setTimeout(() => {
+        expect(dispatchedBrowserEvent.defaultPrevented).toBe(true);
+
+        done();
+      }, 0);
+    });
+
+    test('should end dragging state', () => {
+      const draggable = sandbox.querySelector('li');
+      document.elementFromPoint = () => draggable;
+
+      // Fire dragstart event to initialize dragging state
+      triggerEvent(draggable, 'dragstart', {
+        dataTransfer: getDataTransferStub(),
+      });
+
+      // Fire primary dragend event
+      triggerEvent(draggable, 'dragend');
+
+      expect(dragSensor.dragging).toBe(false);
+    });
+  });
+
+  describe('#_onMouseUp', () => {
+    test('should clear mousedown timeout on mouseup', (done) => {
+      const dragSensor = new DragSensor();
+      const mouseUpEvent = new MouseEvent('mouseup');
+
+      // I do not know of a deterministic way to check whether a timeout has
+      // been cleared, so we must mock the global function.
+      const spy = jest.spyOn(window, 'clearTimeout');
```

Comment:
I think rather than mocking the `clearTimeout` here, we should let it clear the timeout and check that `.dragging` was reset back to `false`, something along the lines of:

```js
triggerEvent(element, mouseDownEvent);
triggerEvent(element, mouseUpEvent);

expect(dragSensor.dragging).toBe(false);
```